tests/testenv/mod.rs

```diff
@@ -186,12 +203,47 @@ impl TestEnv {
         }
 
         // Normalize both expected and actual output.
-        let expected = normalize_output(expected, true);
-        let actual = normalize_output(&String::from_utf8_lossy(&output.stdout), false);
+        let expected = normalize_output(expected, true, self.normalize_line);
+        let actual = normalize_output(
+            &String::from_utf8_lossy(&output.stdout),
+            false,
+            self.normalize_line,
+        );
 
         // Compare actual output to expected output.
         if expected != actual {
             panic!(format_output_error(args, &expected, &actual));
         }
     }
+
+    /// Assert that calling *fd* with the specified arguments produces the expected error.
+    pub fn assert_error(&self, args: &[&str], expected: &str) {
+        self.assert_error_subdirectory(".", args, expected)
+    }
+
+    /// Assert that calling *fd* in the specified path under the root working directory,
+    /// and with the specified arguments produces an error with the expected message.
+    fn assert_error_subdirectory<P: AsRef<Path>>(&self, path: P, args: &[&str], expected: &str) {
+        // Setup *fd* command.
+        let mut cmd = process::Command::new(&self.fd_exe);
+        cmd.current_dir(self.temp_dir.path().join(path));
+        cmd.args(args);
+
+        // Run *fd*.
+        let output = cmd.output().expect("fd output");
+
+        // Check for exit status.
+        if output.status.success() {
+            panic!(
+                "fd exited successfully. Expected error {} did not occur.",
+                expected
+            );
+        }
+
+        // Compare actual output to expected output.
+        let actual = String::from_utf8_lossy(&output.stderr);
+        if expected.len() <= actual.len() && expected != &actual[..expected.len()] {
```

Comment:
Why not just `expected != actual`? Is this stripping the `[fd error]` prefix?

Reply:
The errors regarding the conflict between `exec` and `exec-batch` is generated by `clap`… With the usage message.
This allows to have just the error message in the test.

Reply:
That would actually be `start_with` (as usage is after the message). And it's indeed cleaner!